campaignion_tracking/tests/TrackerTest.php

```diff
@@ -110,17 +115,49 @@ public function testNodeContextDonation() {
   public function testNodeContextRedirectCodes() {
     $redirect = new stdClass();
     $redirect->fragment = '';
-    $submission = new stdClass();
-    $submission->node = $this->petition;
-    $submission->node->title = "Donation test & stuff";
-    $submission->sid = 1;
+    $this->petition->title = "Donation test & stuff";
+    $submission['sid'] = 1;
+    $submission = new Submission($this->petition, (object) $submission);
+    $submission->opt_in = new Values($submission);
     $nid = $submission->node->nid;
+
     campaignion_tracking_webform_redirect_alter($redirect, $submission);
+
     $this->assertEqual('t:t=s;w:nid=' . $nid . '&sid=1&title=Donation%20test%20%26%20stuff', $redirect->fragment);
 
     $redirect->fragment = 'something';
+
     campaignion_tracking_webform_redirect_alter($redirect, $submission);
+
```

Comment:
I don’t think having an empty line after every line helps readability. You could group it into 3 parts: submission creation, creation of the redirect object + first assert, change of the redirect + second assert; each with an empty line it between.

Reply:
I wanted to split it by: setup/change → call → assert. This is indeed a bit weird if changing the redirect + assert is separated.

Reply:
But I agree: putting the redirect object and the call closer together will improve readability

Reply:
Better now?